support-frontend/test/selenium/util/PostDeployTestUser.scala

```diff
@@ -24,3 +28,31 @@ class PostDeployTestUser(driverConfig: DriverConfig) extends TestUser {
   val username = testUsers.generate()
   addTestUserCookies(username)
 }
+
+class PostDeployTestUserSubs(driverConfig: DriverConfig) extends TestUser {
+
+  private val client = new OkHttpClient()
+  private val requestToIdapi = new Request.Builder()
+    .url(Config.idapiNewTestUserUrl)
+    .addHeader(Config.idapiClientAccessTokenName, Config.idapiClientAccessTokenSecret)
+    .post(RequestBody.create(MediaType.parse("application/json"), "{}"))
+    .build()
+  private val response = client.newCall(requestToIdapi).execute()
+  private val responseBodyAsJson = parse(response.body().string()).getOrElse(Json.Null)
+  private val cursor = responseBodyAsJson.hcursor
+
+  private def addTestUserCookies(testUsername: String) = {
+    driverConfig.addCookie(name = "pre-signin-test-user", value = testUsername)
+    driverConfig.addCookie(name = "_test_username", value = testUsername, domain = Some(Config.guardianDomain))
+    driverConfig.addCookie(name = "_post_deploy_user", value = "true") // This enables the tests to use the mocked payment services
+    driverConfig.addCookie(name = "GU_TK", value = "1.1") //To avoid consent banner, which messes with selenium
+    for (n <- 0 until 3) {
```

Comment:
Once you have an object structure for the response via Circe you can `.forEach` over the `values` array to add them. That will mean that if a cookie is added or removed this code won't need to change.

Reply:
👍